kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/DeploymentUtils.java

```diff
@@ -277,4 +278,21 @@ public void conditionEvaluated(EvaluatedCondition<PodStatus> condition) {
             // unused
         }
     }
+
+    /**
+     * Gets node port service url.
+     *
+     * @param namespace the namespace
+     * @param serviceName the service name
+     * @return the node port service url
+     */
+    public static String getNodePortServiceUrl(String namespace, String serviceName) {
+        var nodeIP = kubeClient(namespace).getClient().nodes().list().getItems().get(0).getStatus().getAddresses().get(0).getAddress();
+        var spec = kubeClient().getService(namespace, serviceName).getSpec();
+        int port = spec.getPorts().stream().map(ServicePort::getNodePort).findFirst()
+                .orElseThrow(() -> new KubeClusterException("Unable to get the service port of " + serviceName));
+        String url = nodeIP + ":" + port;
```

Comment:
nit: it is not really a url.  urls have a scheme part.  this is an address or hostPort